### PR TITLE
Homework1

### PR DESCRIPTION
**Pull request by marcindus**
Fix after review
team: marcindus, adamvm
solution also for Narzedzia#3

### CODE REVIEW

File: module2/presentation_virtual.md [hunk 1]
```diff
@@ -175,39 +137,32 @@ ___
...
+```
+
 ___
 
 ## Zadanie 2
 
-Wykorzystując wspólną klasę bazową `Cargo` spróbuj przechowywać wszystkie towary w jednym wektorze w klasie `Ship`.
-
-Dodaj funkcję `void load(std::shared_ptr<Cargo> cargo)`, która dodaje towar i (dla chętnych) `void unload(Cargo* cargo)`, która usuwa towar z obiektu klasy `Ship`.
+Wykorzystując wspólną klasę bazową `Cargo` spróbuj przechowywać wszystkie towary w jednym wektorze w klasie `Ship`. Dodaj funkcję `void load(std::shared_ptr<Cargo> cargo)`, która dodaje towar i (dla chętnych) `void unload(Cargo* cargo)`, która usuwa towar z obiektu klasy `Ship`.
```
Comment (WBigaj) [inline module2/presentation_virtual.md:168]: Why have you added all those files?

File: shm/Player.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <memory>
+#include "Cargo.hpp"
+#include "Ship.hpp"
```
Comment (WBigaj) [inline shm/Player.hpp:5]: There should be newline between memory includes and other file includes.

File: shm/Ship.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+{
+    if (crew_ >= num)
+        crew_ -= num;
+    return *this;
+}
+
+Ship& Ship::operator+=(size_t num)
+{
+    crew_ += num;
+    return *this;
```
Comment (WBigaj) [inline shm/Ship.cpp:30]: How are you checking if you have space in your ship for crewmen you want to add?

File: shm/test.cpp [hunk 4]
```diff
@@ -0,0 +1,100 @@
+#include <vector>
+#include <set>
+
+#include "Cargo.hpp"
+#include "Island.hpp"
+#include "Player.hpp"
+#include "Map.hpp"
+#include "gtest/gtest.h"
```
Comment (WBigaj) [inline shm/test.cpp:8]: Includes should be sorted alphabetically.

**Review by WBigaj — COMMENTED** [review]
I have checked your code and given my feedback.

File: shm/Player.hpp (hunk 2 above)
Comment (ziobron) [reply shm/Player.hpp:5]: It's a matter of convention and if they don't want they don't need to stick to it.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -1,17 +1,29 @@
...
+       working-directory: shm/build
+       run: |
+        cmake ..
+        make
+     - name: Run tests
+       working-directory: shm/build
+       run: ctest -V
+
 
-# In case of conflict do not remove above jobs. Just add your jobs below and remove this comment :)
```
Comment (ziobron) [inline .github/workflows/main.yml:18]: Hello, have you read this comment? :D

File: shm/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+# later. Otherwise we have to add them here ourselves.
+if (CMAKE_VERSION VERSION_LESS 2.8.11)
+  include_directories("${gtest_SOURCE_DIR}/include")
+endif()
+
+project(shm)
+enable_testing()
+
+add_executable(${PROJECT_NAME} main.cpp    ${SHM_SOURCES} )
+add_executable(${PROJECT_NAME}-ut test.cpp ${SHM_SOURCES} )
```
Comment (ziobron) [inline shm/CMakeLists.txt:44]: There is no library for common files

File: shm/Player.hpp [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+class Player
+{
+public:
+    Player(Ship ship, size_t money, size_t space);
+
+    [[nodiscard]] size_t getAvailableSpace() const;
+    [[nodiscard]] size_t getMoney() const;
+    [[nodiscard]] size_t getSpeed() const;
+    [[nodiscard]] Cargo* getCargo(size_t index) const;
```
Comment (ziobron) [inline shm/Player.hpp:15]: [[nodiscard]] - nice 👍

File: shm/test.cpp [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+TEST(cargo, cargoShuldNotBeZero)
```
Comment (ziobron) [inline shm/test.cpp:10]: the name of this test case is quite the opposite of what the test is checking

**Review by ziobron — CHANGES_REQUESTED** [review]
I guess you had a problem with a merge. You have chosen only your files version and diff is showing many unnecessary things. You can always try a rebase with merge - `git pull --rebase`. It will spare you a lot of unnecessary conflicts.

File: shm/Map.hpp [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+#include <random>
+#include <vector>
+
+#include "Island.hpp"
+
+class Map
+{
+
+public:
+    Map() : mapXsize(10), mapYsize(11) { randomIslandGenerator(10); }
```
Comment (DamiXioR) [inline shm/Map.hpp:14]: In my opinion it's better to use constexpr  instead literals 10 and 11.

File: shm/Map.hpp [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    void randomIslandGenerator(int amount)
+    {
+        static std::random_device rd;
+        static std::mt19937 gen(rd());
+
+        std::vector<int> map_(mapXsize * mapYsize);
+        std::iota(std::begin(map_), std::end(map_), 0);
+        std::shuffle(std::begin(map_), std::end(map_), gen);
+
+        for (int i = 0; i < amount; i++)
```
Comment (DamiXioR) [inline shm/Map.hpp:38]: U can use also std::size_t for amount and i

File: shm/Player.hpp (hunk 2 above)
Comment (DamiXioR) [reply shm/Player.hpp:5]: You should move to top headers

**Review by DamiXioR — COMMENTED** [review]
Some ideas

File: .github/workflows/main.yml (hunk 5 above)
Comment (marcindus) [reply .github/workflows/main.yml:18]: yes, i've messed up with main.yml, and really I do not remember how I ended up with this merge... I can give my word that it was not on purpose. :)

**Pull request closed without merging**